src/infra/main/template.yaml

```diff
@@ -996,27 +1044,49 @@ Resources:
         Owner: !Ref OwnerTagValue
         Source: !Ref SourceTagValue
 
-  # PrivateAPIGatewayAccessLogGroup:
-  #   Type: AWS::Logs::LogGroup
-  #   Properties:
-  #     LogGroupName: !Sub "/aws/apigateway/${PrivateRestApi}/${ApiStageName}"
-  #     RetentionInDays: 30
-  #     KmsKeyId: !GetAtt CloudWatchEncryptionKey.Arn
-  #     Tags:
-  #       - Key: Product
-  #         Value: !Ref ProductTagValue
-  #       - Key: System
-  #         Value: !Ref SystemTagValue
-  #       - Key: Environment
-  #         Value: !Ref Environment
-  #       - Key: Name
-  #         Value: PrivateApiGatewayAccessLogGroup
-  #       - Key: Owner
-  #         Value: !Ref OwnerTagValue
-  #       - Key: Source
-  #         Value: !Ref SourceTagValue
-  #       - Key: Name
-  #         Value: !Sub "/aws/apigateway/${PrivateRestApi}/${ApiStageName}"
+  PrivateAPIGatewayAccessLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/apigateway/${PrivateRestApi}/${ApiStageName}"
+      RetentionInDays: 30
+      KmsKeyId: !GetAtt CloudWatchEncryptionKey.Arn
+      Tags:
+        - Key: Product
+          Value: !Ref ProductTagValue
+        - Key: System
+          Value: !Ref SystemTagValue
+        - Key: Environment
+          Value: !Ref Environment
+        - Key: Name
+          Value: PrivateApiGatewayAccessLogGroup
+        - Key: Owner
+          Value: !Ref OwnerTagValue
+        - Key: Source
+          Value: !Ref SourceTagValue
+        - Key: Name
+          Value: !Sub "/aws/apigateway/${PrivateRestApi}/${ApiStageName}"
+
+  PrivateAPIAccessLogGroupNameSSM:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Description: The ARN of the PrivateAPIAccessLogGroup
+      Name: !Sub /${AWS::StackName}/PrivateRestAPI/Access/LogGroup/Name
+      Type: String
+      Value: !Sub /aws/apigateway/${PrivateRestApi}/${ApiStageName}
+      Tags:
+        Environment: !Ref Environment
+        Product: !Ref ProductTagValue
+        System: !Ref SystemTagValue
+        Owner: !Ref OwnerTagValue
+        Source: !Ref SourceTagValue
+
+  PrivateAPIGatewayLogsSubscriptionFilterCSLS:
+    Condition: IsProdOrInt
+    Type: AWS::Logs::SubscriptionFilter
+    Properties:
+      DestinationArn: !Ref CSLSSubscriptionEndpointArn
+      LogGroupName: !Ref PrivateAPIGatewayAccessLogGroup
+      FilterPattern: ""
```

Comment:
is it possible to define the resource without this property? I am not sure how an empty string would be interpreted here

Reply:
Happy enough to remove this if needed, copied line 2837 where the resource for the CSLS subscription uses an empty string as a filter pattern, for the vc storage private api access logs.

Reply:
turns out `FilterPattern` is required from the linter.